webroot/results/includes/_database.php

```diff
@@ -129,50 +122,57 @@ function dbCommand ( $command ) {
 #----------------------------------------------------------------------
 function getAllEvents () {
 #----------------------------------------------------------------------
-  global $cachedEvents;
-  return $cachedEvents;
+  return dbQuery("SELECT * FROM Events WHERE rank<990 ORDER BY rank");
 }
 
 #----------------------------------------------------------------------
 function getAllRounds () {
 #----------------------------------------------------------------------
-  global $cachedRounds;
-  return $cachedRounds;
+  return dbQuery("SELECT * FROM RoundTypes ORDER BY rank");
 }
 
 #----------------------------------------------------------------------
 function getAllCompetitions () {
 #----------------------------------------------------------------------
-  global $cachedCompetitions;
-  return $cachedCompetitions;
+  return dbQuery("
+    SELECT id, cellName
+    FROM Competitions
+    ORDER BY (start_date BETWEEN DATE_SUB(NOW(), INTERVAL 7 DAY) AND DATE_ADD(NOW(), INTERVAL 7 DAY)) DESC, year DESC, month DESC, day DESC
+  ");
 }
 
 #----------------------------------------------------------------------
 function getAllUsedCountries () {
 #----------------------------------------------------------------------
-  global $cachedUsedCountries;
-  return $cachedUsedCountries;
+  return dbQuery("
+    SELECT DISTINCT country.*
+    FROM Results result, Countries country
+    WHERE country.id = countryId
+    ORDER BY country.name
+  ");
 }
 
 #----------------------------------------------------------------------
 function getAllUsedCountriesCompetitions () {
 #----------------------------------------------------------------------
-  global $cachedUsedCountriesCompetitions;
-  return $cachedUsedCountriesCompetitions;
+  return dbQuery("
+    SELECT DISTINCT country.*
+    FROM Competitions competition, Countries country
+    WHERE country.id = countryId
+    ORDER BY country.name
+  ");
 }
 
 #----------------------------------------------------------------------
 function getAllUsedContinents () {
 #----------------------------------------------------------------------
-  global $cachedUsedContinents;
-  return $cachedUsedContinents;
+  return dbQuery("SELECT * FROM Continents");
```

Comment:
So far there's been [an expensive query](https://github.com/thewca/worldcubeassociation.org/pull/1591/files#diff-cbcbe0a5128fb29a366b2533170270aeL46) for that, but obviously all continents are *used*!